### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by waelelsafty07**
## PR Details
- Create an initialState variable for your booksSlice
- Display your books, received from the slice, in a reusable component
- Dispatch actions using useDispatch
- Add a <Button> component, which includes:
  - An event handler that adds a book to the books array (with attributes id, title and author)
- Add a <Button> component, which includes:
  - An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/components/add/Add.js [hunk 1]
```diff
@@ -1,24 +1,39 @@
...
-  <>
+const AddNewBook = () => {
+  const dispatch = useDispatch();
+  const titleRef = useRef(null);
+  const authorRef = useRef(null);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const title = titleRef.current?.value;
+    const author = authorRef.current?.value;
```
Comment (divinecharlotte) [inline src/components/add/Add.js]: - [x] Kindly consider adding the local state for title and author, which is updated through the onChange event to fullfill the below requirements 
> Add a <Button> component, which includes:
- > An event handler that adds a book to the books array (with attributes id, title and author)




hint 👇 

```
import React from "react"
import { useDispatch } from "react-redux"
import { addBook } from "path/to/booksSlice"

function NewBookForm() {
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")
  const books = useSelector((state) => state.books.books)
  const dispatch = useDispatch()

  // Other logic that controls form state

  return (
    <div>
      {/* Rest of form JSX */}
      <Button onClick={() => dispatch(addBook({ title, author }))}>
        Add book
      </Button>
    </div>
  )
}

export default NewBookForm

```

while doing so consider fixing this issue when I add a book the title and author are not being fixed check the image below for more clarification
Screen Shot 2023-06-16 at 18 26 14

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @waelelsafty07   , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## To Highlight 💯 

- Use of GitHub flow👍 
- Descriptive pr 👍 

 




 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/add/Add.js (hunk 1 above)
Comment (waelelsafty07) [reply src/components/add/Add.js]: @divinecharlotte  thank you

**Review by souad988 — APPROVED** [review]
Hi @waelelsafty07 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Good file structure 
Use components for reusable elements
Good communication with reviewer

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**